feat: 마이페이지 구조 정리(Firebase 제거) 및 프로필 수정 기능 추가

whozbebop opened the pull request:
## 변경 사항

### 1. 강좌 시청 페이지 (Learn Page) 안정화
- `useCourseLearn` 훅 정리
  - 강좌 상세 / 수강 정보 / 진도 조회 흐름 명확화
  - learnService 단일 진입점으로 통합하여 service 혼선 제거
- **내 수강목록 조회 기반 enrollmentId 매핑 로직 추가**
  - 수강 중인 강좌 목록 조회 후 courseId → enrollmentId 매칭
  - 수강 여부에 따라 진도 조회 및 UI 분기 처리
- mock / real API 전환 구조 정리

### 2. 강사 강좌 목록 페이지 API 연동 기반 구성
- 강사 강좌 목록 조회용 API service 추가
- 도메인 타입 정의 (`instructor course`)
- mock 데이터 추가로 API 미연동 상태에서도 UI 확인 가능
- 강사 강좌 목록 페이지 기본 구조 연결

### 3. 프로필 닉네임 인라인 수정 기능 추가
- 마이페이지에서 닉네임 직접 수정 가능
- 수정/저장 토글 UI 적용
- 저장 중 상태 처리 및 실패 시 예외 처리
- API 호출 후 즉시 화면 반영

---

## 구조 개선 포인트
- course / enrollment / instructor 도메인 책임 분리
- learnService 중심으로 강좌 시청 관련 API 흐름 통합
- Firebase 의존 제거 후 API 기반 구조로 전환 준비 완료

---

## 리뷰 포인트
- 강좌 시청 페이지 데이터 흐름 (course → enrollment → learn)
- mock / real API 전환 방식 적절성
- 강사 강좌 목록 API 타입 및 구조 적절성

---


close #69

whozbebop committed:
refactor: enrollment 도메인 구조 정리 및 중복 서비스 제거

whozbebop committed:
feat: 강좌 시청 페이지 데이터 흐름 및 mock 연동 정리

whozbebop committed:
feat: 프로필 닉네임 인라인 수정

whozbebop committed:
feat: 강사 강좌 목록 페이지 API 연동 기반 세팅

whozbebop committed:
chore: instructorService 위치 변경, 해당 파일 삭제

whozbebop committed:
chore: mock 사용 비활성화

kimsuyeon0916 commented on src/domains/user/services/userService.ts:
```diff
@@ -36,3 +36,13 @@ export const getProfile = async ({
+  return getApi('/api/users/me', {
+    method: 'PATCH',
+    body: JSON.stringify(payload),
+  });
```
```suggestion
```

제거하고 updateProfile 사용하세요

kimsuyeon0916 commented on src/domains/user/pages/ProfileClientPage.tsx:
```diff
@@ -5,9 +5,42 @@ import Link from 'next/link';
+    try {
+      setIsSaving(true);
+
+      await updateMyProfile({ nickname });
```
```suggestion
      await updateProfile({ nickname });
```

kimsuyeon0916 commented on src/domains/user/pages/ProfileClientPage.tsx:
```diff
@@ -5,9 +5,42 @@ import Link from 'next/link';
 import { formatUserDate } from '@/domains/user/utils/formatUserDate';
 import { useAuthState } from '@/domains/auth/hooks/useAuthState';
+import { useEffect, useState } from 'react';
+import { updateMyProfile } from '../services/userService';
```
```suggestion
import { updateProfile } from '../services/userService';
```

kimsuyeon0916 commented on src/app/courses/[id]/learn/page.tsx:
```diff
@@ -1,5 +1,5 @@
 import CourseLearnClient from '@/domains/course/pages/CourseLearnClientPage';
 
-export default async function CourseLearnPage({ params }: { params: { id: string } }) {
```
되돌리기

kimsuyeon0916 commented on src/domains/course/services/instructorCourseService.ts:
```diff
@@ -0,0 +1,18 @@
+ */
+export async function getInstructorCourses(): Promise<InstructorCoursePage> {
+  if (USE_MOCK) return MOCK_INSTRUCTOR_COURSES;
+
```
```suggestion
/**
 * 내가 등록한 강좌 목록 조회
 * GET /api/instructor/courses
 */
export async function getInstructorCourses(): Promise<InstructorCoursePage> {
```

whozbebop committed:
refactor: PR 리뷰 반영 (공통 profile 수정 함수 적용)

- userService 내 공통 profile 수정 함수 사용
- Next.js app router 규칙에 맞게 page params 시그니처 추가

kimsuyeon0916 commented on src/domains/user/pages/InstructorClientPage.tsx:
```diff
@@ -1,89 +1,58 @@
-                >
-                  삭제
-                </button>
+          <Link href={`/courses/${course.courseId}`}>
```
```suggestion
          <Link href={`/courses/${course.courseId}`} key={course.courseId}>
```

kimsuyeon0916 commented on src/domains/user/pages/InstructorClientPage.tsx:
```diff
@@ -1,89 +1,58 @@
-              <div className={styles["authored__meta"]}>
-                <h3 className={styles["authored__title"]}>{course.title}</h3>
-                <p className={styles["authored__category"]}>
-                  {Array.isArray(course.category)
-                    ? course.category.join(" / ")
-                    : course.category ?? "카테고리 없음"}
-                </p>
-              </div>
-
-              <div className={styles["authored__actions"]}>
-                <button className={styles["authored__btn"]}>수정</button>
-                <button
-                  type="button"
-                  className={`${styles["authored__btn"]} ${styles["authored__btn--delete"]}`}
-                >
-                  삭제
-                </button>
```
👀 히스토리) 추후에 초안 수정/삭제 기능 추가 시, 활용해야 함

kimsuyeon0916 commented on src/domains/user/pages/InstructorClientPage.tsx:
```diff
@@ -1,89 +1,58 @@
+import type { InstructorCourse } from '@/domains/course/types/instructor';
 
-export default function InstructorCoursesPage() {
-  const { user, loading: userLoading } = useAuthState();
```
살리고

kimsuyeon0916 commented on src/domains/user/pages/InstructorClientPage.tsx:
```diff
@@ -1,89 +1,58 @@
-        setCourses(data);
-      } catch (err) {
-        console.error("❌ 강좌 가져오기 실패:", err);
+        const page = await getInstructorCourses();
```
```suggestion
        const page = await getInstructorCourses();
```

여기만 바꾸고, 나머지 다 돌리세요

kimsuyeon0916 commented on src/domains/course/services/learnService.ts:
주석 다시 살리기

kimsuyeon0916 commented on src/domains/course/services/instructorCourseService.ts:
```diff
@@ -0,0 +1,12 @@
+
+/**
+ * 내가 등록한 강좌 목록 조회
+ * GET /api/instructor/courses
```
```suggestion
```

kimsuyeon0916 replied:
```suggestion
```

kimsuyeon0916 commented on src/domains/user/services/enrollmentService.ts:
👀 히스토리용) 수강 단건 조회/진도 조회 -> 아직 미사용
**수강 목록 조회 500 에러남**

kimsuyeon0916 commented on src/domains/user/pages/EnrollmentClientPage.tsx:
```diff
@@ -0,0 +1,76 @@
+
+import { useEffect, useState } from 'react';
+import Link from 'next/link';
+
```
```suggestion
```

whozbebop committed:
chore: instructor 페이지 원복 및 강의 조회 호출 방식 정리

kimsuyeon0916 commented on src/domains/user/types/enrollment.ts:
서비스 함수에 쓰이는 함수들 타입은 요청은 Request, 응답은 Response 꼭 붙여주세요!! 안그럼 헷갈립니다.

kimsuyeon0916 commented on src/domains/course/types/instructor.ts:
서비스 함수에 쓰이는 함수들 타입은 요청은 Request, 응답은 Response 꼭 붙여주세요!! 안그럼 헷갈립니다.

kimsuyeon0916 commented on src/domains/course/hooks/useCourseLearn.ts:
👀 히스토리용) 강좌 별 수강 정보 조회 (수강 여부 확인 by Course Id) 완성되면, 사실상 롤백해야함

kimsuyeon0916 replied:
@whozbebop

kimsuyeon0916 commented on src/domains/course/utils/mapCourse.ts:
서비스 함수에 쓰이는 함수들 타입은 요청은 Request, 응답은 Response 꼭 붙여주세요!! 안그럼 헷갈립니다.

kimsuyeon0916 commented on src/domains/course/types/learn.ts:
서비스 함수에 쓰이는 함수들 타입은 요청은 Request, 응답은 Response 꼭 붙여주세요!! 안그럼 헷갈립니다.

whozbebop committed:
refactor: learn/instructor 타입 네이밍 정리 및 미사용 파일 제거

whozbebop committed:
Merge branch 'develop' into feature/#69

whozbebop committed:
refactor: instructor 타입 네이밍 정리 및 주석 삭제
